app/src/features/authentication/signup/ConfirmationEmailModal.tsx

```diff
@@ -0,0 +1,168 @@
+import styled from "styled-components";
+import { useState } from "react";
+import CodeInputField from "../../../components/CodeInputField";
+import { UseVerifyEmail } from "./hooks/useVerifyEmail";
+import { useNavigate } from "react-router-dom";
+import { UseSendConfirmationEmail } from "./hooks/useSendConfirmationEmail";
+
+const ModalOverlay = styled.div`
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background: rgba(0, 0, 0, 0.5);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 1000;
+`;
+
+const ModalContainer = styled.div`
+  position: relative;
+  background: white;
+  padding: 2rem;
+  border-radius: 8px;
+  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
+  text-align: center;
+  width: 400px;
+`;
+
+const Button = styled.button<{ disabled?: boolean }>`
+  border: none;
+  border-radius: var(--border-radius-default-tiny);
+  padding: 0.6rem 1rem;
+  margin: 1rem;
+  background-color: ${({ disabled }) =>
+    disabled ? "transparent" : "var(--accent-color)"};
+  color: ${({ disabled }) =>
+    disabled ? "var( --color-text-secondary)" : "var(--color-text-button)"};
+  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
+  opacity: ${({ disabled }) => (disabled ? 0.6 : 1)};
+
+  &:hover {
+    background-color: "var(--color-item-hover)";
+  }
+`;
+const ModalTitle = styled.h2`
+  margin-bottom: 1rem;
+  font-color: (--accent-color: #56a2c9);
+`;
+
+const ModalMessage = styled.p`
+  margin-bottom: 2rem;
+`;
+
+type ConfirmationEmailModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  email: string;
+};
+const CloseButton = styled.button`
+  position: absolute;
+  top: 1rem;
+  right: 1rem;
+  background: transparent;
+  border: none;
+  font-size: 2rem;
+  color: var(--accent-color);
+  cursor: pointer;
+
+  &:hover {
+    color: var(--color-item-hover);
+  }
+`;
+
+const ResendText = styled.span`
+  color: var(--accent-color);
+  cursor: pointer;
+  text-decoration: underline;
+
+  &:hover {
+    color: var(--color-hover);
+  }
+`;
+const Error = styled.p`
+  align-self: center;
+  color: var(--color-error);
+`;
+const Success = styled.p`
+  align-self: center;
+  color: var(--color-success);
+`;
+
+function ConfirmationEmailModal({
+  isOpen,
+  onClose,
+  email,
+}: ConfirmationEmailModalProps) {
+  const [codeDisplay, setCodeDisplay] = useState<string[]>(Array(6).fill(""));
+  const [error, setError] = useState<string>("");
+  const {
+    verifyCode,
+    isPending: isPendingVerification,
+    isError: isErrorVerification,
+    isSuccess: isVerified,
+  } = UseVerifyEmail();
+  const {
+    SendConfirmationCode,
+    isSuccess: isCodeRensent,
+    isError: isErrorResendingCode,
```

Comment:
Hmmmm..I think this isErrorResendingCode is of no use as you already sets the error state if there's an error and you can use it back down to render the error optionally

Reply:
so you mean we should only have a generalized error ?

Reply:
you already have a state called error and you use setError it an error occurred
so you can use this state only as they do the same job